Undefined Refs in Builds

Im seeing this locally and: https://api.travis-ci.org/v3/job/359973671/log.txt
```
  CCLD     test/helper/tpm_transientempty
test/helper/test_helper_tpm_transientempty-tpm_transientempty.o: In function `main':
/workspace/tpm2-tss/test/helper/tpm_transientempty.c:27: undefined reference to `get_test_opts_from_env'
/workspace/tpm2-tss/test/helper/tpm_transientempty.c:28: undefined reference to `sanity_check_test_opts'
/workspace/tpm2-tss/test/helper/tpm_transientempty.c:31: undefined reference to `sapi_init_from_opts'
/workspace/tpm2-tss/test/helper/tpm_transientempty.c:47: undefined reference to `sapi_teardown_full'
collect2: error: ld returned 1 exit status
Makefile:4330: recipe for target 'test/helper/tpm_transientempty' failed
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -64,6 +64,13 @@ before_script:
   - ./bootstrap
 
 script:
+# build with no tests enabled
+  - mkdir ./build-no-tests
+  - pushd ./build-no-tests
+  - ../configure
+  - make -j$(nproc)
+  - popd
+# build with all tests enabled
   - mkdir ./build
   - pushd ./build
   - |

--- a/Makefile-test.am
+++ b/Makefile-test.am
@@ -48,13 +48,13 @@ test_helper_tpm_startup_CFLAGS = $(TESTS_CFLAGS) -I$(srcdir)/test/integration
 test_helper_tpm_startup_LDFLAGS = $(TESTS_LDFLAGS)
 test_helper_tpm_startup_LDADD = $(TESTS_LDADD)
 test_helper_tpm_startup_SOURCES = test/helper/tpm_startup.c
-endif #SIMULATOR_BIN
 
 noinst_PROGRAMS += test/helper/tpm_transientempty
 test_helper_tpm_transientempty_CFLAGS = $(TESTS_CFLAGS) -I$(srcdir)/test/integration
 test_helper_tpm_transientempty_LDFLAGS = $(TESTS_LDFLAGS)
 test_helper_tpm_transientempty_LDADD = $(TESTS_LDADD)
 test_helper_tpm_transientempty_SOURCES = test/helper/tpm_transientempty.c
+endif #SIMULATOR_BIN
 
 if UNIT
 TESTS_UNIT  = \

